src/param/src/lib.rs

```diff
@@ -128,26 +136,94 @@ impl Param {
 
             // Parse value.
             match key {
-                b"TITLE" => title = Some(Self::read_utf8(&mut raw, i, format, len, 128)?),
-                b"TITLE_ID" => title_id = Some(Self::read_utf8(&mut raw, i, format, len, 12)?),
+                b"APP_VER" => {
+                    app_ver = Some(Self::read_utf8(&mut raw, i, format, len, 8)?);
+                }
+                b"CATEGORY" => {
+                    let category_param = Self::read_utf8(&mut raw, i, format, 4, 4)?;
+                    category = Some(category_param.clone());
+                    if !category_param.contains("bd") && !category_param.contains("ac")  // Blu-Ray Game, DLC.
+                        // Check if this is a patch
+                        && !category_param.starts_with("gp")
+                        // Check if this is an application
+                        && !category_param.starts_with("gd")
+                    {
+                        // For types such as gc, sd, la, and wda, there is no Title or TitleID.
+                        title = Some(format!("No Title {}", category_param).to_string());
+                        title_id = Some("No TitleID".to_string());
```

Comment:
Better to set these to `None` outside this loop if the condition is meet.

Reply:
Reason is so that Discord can show "No Title/No TitleID." It was kinda a prep for later when we add homebrew.

Reply:
By homebrew, I mean .bin and .elf support... Which we should be able to support now.

Reply:
It should be the job of the user of this struct to show those information when these fields is absent.